source/ada/lsp-ada_handlers.adb

```diff
@@ -247,6 +247,13 @@ package body LSP.Ada_Handlers is
       URI  : LSP.Types.LSP_String) return LSP.Types.LSP_String;
    --  Turn URI into path
 
+   function Get_Project
+     (Self : access Message_Handler;
+      Uri  : LSP.Messages.DocumentUri)
+      return GNATCOLL.Projects.Project_Type;
+   --  Retrieve the project that the file belongs to. Root project is returned
```

Comment:
You should say what it does for aggregate projects (that's the purpose of your function)